source/ada/lsp-ada_file_sets.adb

```diff
@@ -81,36 +82,71 @@ package body LSP.Ada_File_Sets is
    --------------------
 
    procedure Get_Any_Symbol
-     (Self        : Indexed_File_Set'Class;
-      Prefix      : VSS.Strings.Virtual_String;
-      Only_Public : Boolean;
-      Callback : not null access procedure
+     (Self              : Indexed_File_Set'Class;
+      Pattern           : LSP.Search.Search_Pattern_Access;
+      Only_Public       : Boolean;
+      Get_Defining_Name : not null access function
         (File : GNATCOLL.VFS.Virtual_File;
-         Loc  : Langkit_Support.Slocs.Source_Location;
-         Stop : in out Boolean))
+         Loc  : Langkit_Support.Slocs.Source_Location)
+      return Libadalang.Analysis.Defining_Name;
+      Callback          : not null access procedure
+        (File          : GNATCOLL.VFS.Virtual_File;
+         Defining_Name : Libadalang.Analysis.Defining_Name;
+         Stop          : in out Boolean))
    is
-      Stop   : Boolean := False;
-      Cursor : Symbol_Maps.Cursor :=
-        Self.All_Symbols.Ceiling (Prefix);
+      Stop          : Boolean := False;
+      Cursor        : Symbol_Maps.Cursor;
+      Defining_Name : Libadalang.Analysis.Defining_Name;
+
    begin
+      if Pattern.Get_Kind = LSP.Messages.Full_Text
+        and then not Pattern.Get_Negate
+        and then Pattern.Get_Whole_Word
+      then
+         Cursor := Self.All_Symbols.Ceiling
```

Comment:
I guess we should always call `Canonicalize`. Consider: we do case insensitive search of 'TEXT', then Ceiling ('TEXT') will not return the position of 'text'.

Reply:
Pattern already contains Canonicalize-ed text when it is not Case_Sensitive, so we do not need to call Canonicalize second time